commit 14c3a9aed852c7383cf67bb90f84ff10eb36cdfd
Author: Evgeni Golov <evgeni@grml.org>
Date:   Wed Mar 6 15:20:46 2013 +0100

    Checking for ${overlay}, not ${device} when trying to close ${overlay}.

diff --git a/scripts/boot/9990-overlay.sh b/scripts/boot/9990-overlay.sh
index 10054e8..c6de171 100755
--- a/scripts/boot/9990-overlay.sh
+++ b/scripts/boot/9990-overlay.sh
@@ -392,7 +392,7 @@ setup_unionfs ()
 		# Close unused overlays (e.g. due to missing $persistence_list)
 		for overlay in ${overlay_devices}
 		do
-			if echo ${used_overlays} | grep -qve "^\(.* \)\?${device}\( .*\)\?$"
+			if echo ${used_overlays} | grep -qve "^\(.* \)\?${overlay}\( .*\)\?$"
 			then
 				close_persistence_media ${overlay}
 			fi
